CRM/Contract/Upgrader.php

```diff
@@ -167,43 +169,82 @@ public function upgrade_2003() {
 
   protected function ensureNoPaymentRequiredPaymentInstrument() {
     try {
-      $optionGroup = civicrm_api3('OptionGroup', 'getsingle', [
-        'name' => 'payment_instrument',
-      ]);
+      $og = OptionGroup::get(FALSE)
+        ->addWhere('name', '=', 'payment_instrument')
+        ->setSelect(['id'])
+        ->setLimit(1)
+        ->execute()
+        ->first();
     }
-    catch (Exception $e) {
+    catch (\Throwable $e) {
+      return;
+    }
+    if (!$og || empty($og['id'])) {
+      return;
+    }
+    $gid = (int) $og['id'];
+
+    $currentNone = OptionValue::get(FALSE)
+      ->addWhere('option_group_id', '=', $gid)
+      ->addWhere('name', '=', 'None')
+      ->setSelect(['id', 'value'])
+      ->setLimit(1)
+      ->execute()
+      ->first();
+
+    $legacy = OptionValue::get(FALSE)
+      ->addWhere('option_group_id', '=', $gid)
+      ->addWhere('name', '=', 'no_payment_required')
+      ->setSelect(['id', 'value'])
+      ->setLimit(1)
+      ->execute()
+      ->first();
+
+    if ($legacy && !$currentNone) {
+      OptionValue::update(FALSE)
+        ->addWhere('id', '=', $legacy['id'])
+        ->addValue('name', 'None')
+        ->addValue('label', 'No payment required')
+        ->execute()
+        ->single();
+      CRM_Core_PseudoConstant::flush();
       return;
     }
 
-    $existing = civicrm_api3('OptionValue', 'get', [
-      'option_group_id' => $optionGroup['id'],
-      'name' => 'None',
-      'return' => 'id',
-    ]);
-    if ($existing['count'] > 0) {
+    if ($legacy && $currentNone) {
+      OptionValue::delete(FALSE)
+        ->addWhere('id', '=', $legacy['id'])
+        ->execute();
+      CRM_Core_PseudoConstant::flush();
       return;
     }
 
-    civicrm_api3('OptionValue', 'create', [
-      'option_group_id' => $optionGroup['id'],
-      'label' => 'No payment required',
-      'name' => 'None',
-      'value' => $this->findNextAvailablePaymentInstrumentValue($optionGroup['id']),
-      'is_active' => 1,
-      'is_reserved' => 0,
-      'weight' => 99,
-    ]);
+    if (!$currentNone) {
+      OptionValue::create(FALSE)
+        ->addValue('option_group_id', $gid)
+        ->addValue('label', 'No payment required')
+        ->addValue('name', 'None')
+        ->addValue('value', $this->findNextAvailablePaymentInstrumentValue($gid))
+        ->addValue('is_active', 1)
+        ->addValue('is_reserved', 0)
+        ->addValue('weight', 99)
+        ->execute()
+        ->single();
```

Comment:
`->single()` doesn't do anything here, as the result of the API call is not assigned to a variable, and can be removed.

Also applies to code further below inside `ensurePaymentInstrumentNone()`.